paasta_tools/cleanup_chronos_jobs.py

```diff
@@ -110,13 +143,17 @@ def main():
     config = chronos_tools.load_chronos_config()
     client = chronos_tools.get_chronos_client(config)
 
-    # get_chronos_jobs_for_cluster returns (service, job)
-    expected_service_jobs = set([chronos_tools.compose_job_id(*job) for job in
-                                 chronos_tools.get_chronos_jobs_for_cluster(soa_dir=args.soa_dir)])
+    running_jobs = set(deployed_job_names(client))
+
+    expected_service_jobs = [chronos_tools.compose_job_id(*job) for job in
+                             chronos_tools.get_chronos_jobs_for_cluster(soa_dir=args.soa_dir)]
+
+    all_tmp_jobs = filter_tmp_jobs(filter_paasta_jobs(running_jobs))
+    expired_tmp_jobs = filter_expired_tmp_jobs(client, all_tmp_jobs)
 
-    running_service_jobs = set(deployed_job_names(client))
+    valid_tmp_jobs = [job for job in all_tmp_jobs if job not in expired_tmp_jobs]
 
-    to_delete = running_service_jobs - expected_service_jobs
+    to_delete = [job for job in running_jobs if job not in expected_service_jobs and job not in valid_tmp_jobs]
```

Comment:
If:


```
running_jobs == expected_service_jobs + all_tmp_jobs + something_else_that_needs_removal
```

and:

```
all_tmp_jobs == valid_tmp_jobs + expired_tmp_jobs
```

any reason for not making everything a set and doing something like (arguably more readable):

```
valid_tmp_jobs = all_tmp_jobs - expired_tmp_jobs
to_delete = running_jobs - expected_service_jobs - valid_tmp_jobs
```

What's left should be: `something_else_that_needs_removal` + `expired_tmp_jobs`.